src/wikibase.py

```diff
@@ -267,58 +272,123 @@ def get_concept(
             concept.alternative_labels = list(all_positive_labels)
             concept.negative_labels = list(all_negative_labels)
 
+        if include_ancestry:
+            concept.ancestry = self.get_ancestry(wikibase_id)
+
         return concept
 
-    def get_concept_ids(self) -> list[WikibaseID]:
+    def get_ancestry(self, wikibase_id: WikibaseID) -> list[WikibaseID]:
+        """Fetch the complete ancestry of a concept"""
+        # Resolve any redirects first
+        wikibase_id = self._resolve_redirect(wikibase_id)
+
+        response = self.session.get(
+            url=self.api_url,
+            params={
+                "action": "wbgetentities",
+                "format": "json",
+                "ids": wikibase_id,
+                "props": "claims",
+            },
+        ).json()
+
+        entity = response["entities"][wikibase_id]
+        ancestry = [wikibase_id]
+        if "claims" in entity:
+            for claim in entity["claims"].values():
+                for statement in claim:
+                    if statement["mainsnak"]["snaktype"] == "value":
+                        property_id = statement["mainsnak"]["property"]
+                        value = statement["mainsnak"]["datavalue"]["value"]
+                        if property_id == self.subconcept_of_property_id:
+                            resolved_id = self._resolve_redirect(value["id"])
+                            ancestry.append(resolved_id)
+                            ancestry.extend(self.get_ancestry(resolved_id))
+
+        # Duplicates can occur in the ancestry, so they need to be removed
+        return list(set(ancestry))
+
+    def _get_pages(self, extra_params: dict) -> list[dict]:
         """
-        Get concept ids from Wikibase.
+        Helper method to get pages from Wikibase with pagination.
 
-        :return list[WikibaseID]: The concept ids, e.g ["Q123", "Q456"]
+        :param dict extra_params: The parameters to pass to the API
+        :return list[dict]: List of page data from all batches
         """
         PAGE_REQUEST_SIZE = 500
-        # An extra precaution against infinite loops, suitable up to 1M concepts (500*2000)
-        MAX_PAGE_REQUESTS = 2000
+        MAX_PAGE_REQUESTS = 2000  # Suitable up to 1M pages (500*2000)
 
-        params = {
+        base_params = {
             "action": "query",
             "format": "json",
-            "list": "allpages",  # See https://www.mediawiki.org/wiki/API:Allpages
+            "list": "allpages",
```

Comment:
The link to API docs was removed from this. Personally I found it useful as Wikibase docs are hard to navigate but it could also live in the docstring? Wdyt?

Reply:
good point - will re-add